### PR TITLE
Include co-taught sections in sections#index API response.

### PR DESCRIPTION
**Pull request by etaderhold**
This change will allow teachers to see sections they co-teach along with sections they own, on the dashboard homepage and elsewhere that calls this API endpoint.

## Links
- [Co-teacher spec](https://docs.google.com/document/d/1zyODkc7VXsAb17W3jefNM9JNkQVZ8nU2q5C9Xlp7cZs/edit#heading=h.3hotlme3tb4g)
- [JIRA](https://codedotorg.atlassian.net/browse/TEACH-72)

## Testing story

Automated test included to ensure that owned and co-taught sections are each included exactly once in the response.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -471,6 +471,16 @@ def self.find_or_create_facilitator(params, invited_by_user)
   has_many :followers, through: :sections
   has_many :students, through: :followers, source: :student_user
 
+  has_many :section_instructors, foreign_key: 'instructor_id'
+  has_many :active_section_instructors, -> {where(status: :active)}, class_name: 'SectionInstructor', foreign_key: 'instructor_id'
+  has_many :sections_instructed, through: :active_section_instructors, source: :section
+
+  # TODO once the backfill creates SectionInstructors for every section owner,
```
Comment (lfryemason) [inline dashboard/app/models/user.rb:478]: Could we create a JIRA task to remove this in a week or so? Just don't want it to get lost and forgotten

File: dashboard/test/controllers/api/v1/sections_controller_test.rb [hunk 2]
```diff
@@ -84,6 +84,18 @@ class Api::V1::SectionsControllerTest < ActionController::TestCase
     assert_equal expected, returned_json
   end
 
+  test 'returns all sections instructed by teacher' do
+    cotaught_section = create(:section, user: create(:teacher), login_type: 'word')
+    create(:section_instructor, instructor: @teacher, section: cotaught_section, status: :active)
```
Comment (lfryemason) [inline dashboard/test/controllers/api/v1/sections_controller_test.rb:89]: Should we also test that a section where the user is the owner but section_instructor does not exist is still returned?

File: dashboard/app/models/user.rb (hunk 1 above)
Comment (etaderhold) [reply dashboard/app/models/user.rb:478]: Done. https://codedotorg.atlassian.net/browse/TEACH-721

File: dashboard/test/controllers/api/v1/sections_controller_test.rb (hunk 2 above)
Comment (etaderhold) [reply dashboard/test/controllers/api/v1/sections_controller_test.rb:89]: Done. This test will need to be deleted once the post-backfill code cleanup is done, but for now it's important to verify that these old sections still remain in the list.

**Review by lfryemason — APPROVED** [review]

**Pull request merged**